### PR TITLE
fix(pay_ios): allow a lower iOS version than 12 to use the plugin

### PR DESCRIPTION
**Pull request by tudor07**
I have a Flutter app that uses iOS 11 as a minimum target.
By adding this plugin I receive a lot of errors about how some parts of the code require iOS 12.
This can be fixed very easily with a bunch of `#available` checks.
This fixes: https://github.com/google-pay/flutter-plugin/issues/88, https://github.com/google-pay/flutter-plugin/issues/36

### CODE REVIEW

**Comment by tudor07** [comment]
@googlebot I signed it!

File: pay_ios/ios/Classes/ApplePayButtonView.swift [hunk 1]
```diff
@@ -173,10 +173,13 @@ extension PKPaymentButtonType {
     case "donate":
       return .donate
     case "checkout":
+        guard #available(iOS 12.0, *) else { return nil }
```
Comment (JlUgia) [inline pay_ios/ios/Classes/ApplePayButtonView.swift:176]: Check the indentation.

**Comment by JlUgia** [comment]
Also, the target version in the example can be set to anything <12.0 (project.pbxproj). I'd suggest we go with 11.0.

File: pay_ios/ios/Classes/ApplePayButtonView.swift (hunk 1 above)
Comment (tudor07) [reply pay_ios/ios/Classes/ApplePayButtonView.swift]: FIxed in ca9d1002773af4c7a39cf53dfea224761dbb9d30

**Comment by tudor07** [comment]
> Also, the target version in the example can be set to anything <12.0 (project.pbxproj). I'd suggest we go with 11.0.

FIxed in ca9d1002773af4c7a39cf53dfea224761dbb9d30

**Comment by tudor07** [comment]
@JlUgia pushed fixes addressing your comments

**Comment by JlUgia** [comment]
LGTM

**Pull request merged**